packages/js/block-templates/src/hooks/use-layout-template/use-layout-template.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet, natively (not until 7.0.0).
+// Including `@types/wordpress__data` as a devDependency causes build issues,
+// so just going type-free for now.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityRecord } from '@wordpress/core-data';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet, natively (not until 7.0.0).
+// Including `@types/wordpress__data` as a devDependency causes build issues,
+// so just going type-free for now.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useDispatch, useSelect, select as WPSelect } from '@wordpress/data';
+
+export const useLayoutTemplate = ( layoutTemplateId: string | undefined ) => {
+	const layoutTemplateEntity = useSelect( ( select: typeof WPSelect ) => {
+		const { getEntity } = select( 'core' );
+		return getEntity( 'root', 'wcLayoutTemplate' );
```

Comment:
Should we use the getEntityConfig() instead? [getEntity](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/#getentity) has been deprecated:

> Deprecated since WordPress 6.0. Use getEntityConfig instead